src/store/kv-block-store.ts

```diff
@@ -0,0 +1,205 @@
+          stack: error.stack,
+        },
+      );
+    }
+    return false;
+  }
+
+  async hasHeight(height: number): Promise<boolean> {
+    try {
+      const hasHeight = await this.kvBufferStore.has(this.blockHeight(height));
+      return hasHeight;
+    } catch (error: any) {
+      this.log.error(
+        'Failed to verify if block height exists in key/value store',
+        {
+          height,
+          message: error.message,
+          stack: error.stack,
+        },
+      );
+    }
+    return false;
+  }
+
+  async getByHash(hash: string): Promise<PartialJsonBlock | undefined> {
+    try {
+      if (await this.hasHash(hash)) {
+        const blockDataBuffer = await this.kvBufferStore.get(
+          this.blockHash(hash),
+        );
+        if (blockDataBuffer === undefined) {
+          throw new Error('Missing block data in key/value store');
+        }
+        const block = msgpackToJsonBlock(blockDataBuffer);
+        sanityCheckBlock(block);
+        return block;
+      }
+    } catch (error: any) {
+      this.log.error('Failed to get block by hash', {
+        hash,
+        message: error.message,
+        stack: error.stack,
+      });
+    }
+    return undefined;
+  }
+
+  async getByHeight(height: number): Promise<PartialJsonBlock | undefined> {
+    try {
+      if (await this.hasHeight(height)) {
+        const blockHashBuffer = await this.kvBufferStore.get(
+          this.blockHeight(height),
+        );
+        if (blockHashBuffer === undefined) {
+          throw new Error('Missing block hash in key/value store for height.');
+        }
+        const blockHash = toB64Url(blockHashBuffer);
+        return this.getByHash(blockHash);
+      }
+    } catch (error: any) {
+      this.log.error('Failed to get block by height', {
+        height,
+        message: error.message,
+        stack: error.stack,
+      });
+    }
+    return undefined;
+  }
+
+  async delByHash(hash: string): Promise<void> {
+    try {
+      if (await this.hasHash(hash)) {
+        const blockData = await this.getByHash(hash);
+        // remove the hash to block data reference
+        await this.kvBufferStore.del(this.blockHash(hash));
+
+        // remove the block height to hash reference
+        if (
+          blockData !== undefined &&
+          (await this.hasHeight(blockData.height))
+        ) {
+          // TODO: should we check the hash stored in the table matches the blockData, or delete regardless
+          await this.kvBufferStore.del(this.blockHeight(blockData.height));
+        }
+      }
+    } catch (error: any) {
+      this.log.error('Failed to delete block by height', {
+        hash,
+        message: error.message,
+        stack: error.stack,
+      });
+    }
+  }
+
+  async delByHeight(height: number): Promise<void> {
+    try {
+      if (await this.hasHeight(height)) {
+        const blockHashBuffer = await this.kvBufferStore.get(
+          this.blockHeight(height),
```

Comment:
Might as well save a tiny bit of CPU and just call `blockHeight` once.

Reply:
[66f4cf7](https://github.com/ar-io/ar-io-node/pull/60/commits/66f4cf7800807ed7f587e875fbde7653a103bb18)